### ISSUE TITLE
[Remote-SSH Bug]: 

### ISSUE THREAD

**Issue by prager** [microsoft/vscode-remote-release#8182]
### Is there an existing issue for this bug?

- [X] I have searched the existing issues

### Required Troubleshooting Steps

- [X] I have followed these troubleshooting steps
- [X] I have tried both values of the `remote.SSH.useLocalServer` setting

### Connect Locally

It doesn’t connect successfully (same error message)

### ->

When remote.SSH.useLocalServer is checked, there is no error or output in VS Code Terminal, Output, or Problem panes. The green "Openning Remote" indicator in lower left corner is spinning indefinitively and can be stopped only by terminating the connection. The left pane acts like there is a connection, but the button "Open Folder" doesn't do anything. 

Something is clearly broke.

The VS Code Terminal output when remote.SSH.useLocalServer is unchecked is as follows:

```
* Visual Studio Code Server
*
* By using the software, you agree to
* the Visual Studio Code Server License Terms (https://aka.ms/vscode-server-license) and
* the Microsoft Privacy Statement (https://privacy.microsoft.com/en-US/privacystatement).
*
 
Checking server status on port 36533 with wget
e7a6ac040071: start
SSH_AUTH_SOCK====
DISPLAY====
webUiAccessToken====
listeningOn==36533==
osReleaseId==debian==
arch==x86_64==
tmpDir==/tmp==
platform==linux==
unpackResult====
didLocalDownload==0==
downloadTime====
installTime====
extInstallTime====
serverStartTime====
connectionToken==7528167b-3c75-4a3e-af4a-35d4b6a52368==
e7a6ac040071: end
channel 3: open failed: administratively prohibited: open failed
channel 4: open failed: administratively prohibited: open failed

```

### Expected Behavior

Clicking on the "Open Folder" button should open the remote server explorer. The green mode indicator in the lower left corner should stop spinning.

### Steps To Reproduce

1. Click on "Open a Remote Window" green button lower left corner
2. Click on "Add new SSH host"
3. Enter SSH connection command such as ssh username@host.com
4. Edit config file such as:

```
Host MyHost
  HostName myhost.com
  User username
```

5. Click on "Save" and connect to MyHost

### Remote-SSH Log

Output is blank. The status is "Opening Remote..." indefinitively until I terminate the connection.

### Anything else?

The docs are very hard to follow. Sometimes even the English grammar is not correct such as in [https://code.visualstudio.com/docs/remote/troubleshooting#_enabling-alternate-ssh-authentication-methods](https://code.visualstudio.com/docs/remote/troubleshooting#_enabling-alternate-ssh-authentication-methods): 

> ... If you are still having trouble, you may need to the following properties in settings.json and retry:  ... 

It is missing a noun here. By far this is not the only one. There is more, but unfortunately I don't have time at this moment to point out all the problems with the docs.

On the other hand, I realize this is an open source project and it may depend on unpaid volunteers...

**Comment by prager** [comment]
I went through the entire docs and tried other options such as the alternate SSH authentication methods: [Enabling alternate SSH authentication methods](https://code.visualstudio.com/docs/remote/troubleshooting#_enabling-alternate-ssh-authentication-methods) always with the same result

**Comment by prager** [comment]
I didn't realize that I left out my system info:

VS Code version: Code 1.76.1 (5e805b79fcb6ba4c2d23712967df89a089da575b, 2023-03-08T16:31:31.010Z)
OS version: Linux x64 5.15.0-67-generic
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 5500 (12 x 4249)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|0, 0, 0|
|Memory (System)|62.59GB (55.25GB free)|
|Process Argv|--unity-launch --crash-reporter-id a5c163b8-cc6a-4625-bddd-62c844217833|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|cinnamon|
|XDG_CURRENT_DESKTOP|X-Cinnamon|
|XDG_SESSION_DESKTOP|cinnamon|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (4)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-pull-request-github|Git|0.60.0
remote-ssh|ms-|0.98.0
remote-ssh-edit|ms-|0.84.0
remote-explorer|ms-|0.2.0


</details>

**Comment by eleanorjboyd** [comment]
> Click on "Open a Remote Window" green button lower left corner
Click on "Add new SSH host"
Enter SSH connection command such as ssh [username@host.com](mailto:username@host.com)
Edit config file such as:

when you select "connect to host" instead of "add new ssh host" does anything change? It should already be added if you have done it once.

Does the screen get a popup that says something like "failed to connect" with options "close remote", "retry", and "more actions"? If you click "more actions" you should then be able to go to view -> output -> select remote ssh from the dropdown in order to get the full remote-ssh logs. 

Thanks

**Comment by prager** [comment]
No, it doesn't matter if is I select "connect to host" or "add new ssh host" and add the new data. Then there is no indication that something is failing, the green button in lower left corner spins indefinitively with no error message.

**Comment by eleanorjboyd** [comment]
Hello- looked through your logs and saw `channel 3: open failed: administratively prohibited: open failed` so it might be a permissions problem. Do you have tcp port forwarding disabled or any other settings that might interfere with port forwarding?

**Comment by joshspicer** [comment]
One thing that may cause this is our use of dynamic port-forwarding along with `AllowTcpForwarding`, which (at least as of today) is covered in our docs: https://code.visualstudio.com/docs/remote/troubleshooting#_troubleshooting-hanging-or-failing-connections

Please feel free to open a new issue with a current copy of your logs (`> Remote-SSH: Show Log` in the command palette) if you'd like me to take a closer look at any current issues.
